src/Microsoft.ML.Data/Utilities/LocalEnvironment.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.ML.Runtime.Data
+{
+    using Stopwatch = System.Diagnostics.Stopwatch;
+    public sealed class LocalEnvironment : HostEnvironmentBase<LocalEnvironment>
```

Comment:
>LocalEnvironment [](start = 24, length = 16)

Maybe SilentEnvironment? Or you somehow can hook up to the messages?

Reply:
You can hook up via `AddListener<ChannelMessage>`

---
In reply to: [217869228](https://github.com/dotnet/machinelearning/pull/923#discussion_r217869228) [](ancestors = 217869228)